### PR TITLE
Houman lm maf to custom track

### PR DESCRIPTION
**Pull request by HoumanLM**
# Pull Request Checklists

**Important:** When opening a pull request, keep only the applicable checklist and delete all other sections.

### Required

- [x] I tested the new code for my use case (please provide a reproducible example of how you tested the new functionality)

- [ ] I ensured all dplyr functions that commonly conflict with other packages are fully qualified. 

This can be checked and addressed by running `check_functions.pl` and responding to the prompts. Test your code _after_ you do this.

- [ ] I generated the documentation and checked for errors relating to the new function (e.g. `devtools::document()`) and added `NAMESPACE` and all other modified files in the root directory and under `man`. 

- [ ] I have rebuilt the site with `pkgdown::build_site(lazy = TRUE)` to reflect any updated package documentation.

### Optional but preferred with PRs

- [ ] I updated and/or successfully knitted a vignette that relies on the modified code (which ones?)

### Required

- [ ] I documented my function using [ROxygen style](https://jozef.io/r102-addin-roxytags/#:~:text=Inserting%20a%20skeleton%20%2D%20Do%20this,Shift%2BAlt%2BR%20).)

- [ ] Adequate function documentation (see [new-function documentation template](https://github.com/morinlab/GAMBLR#title) for more info)

Example:
```
#' @title ASHM Rainbow Plot
#'
#' @description Make a rainbow plot of all mutations in a region, ordered and coloured by metadata.
#'
#' @details This function creates a rainbow plot for all mutations in a region. Region can either be specified with the `region` parameter,
#' or the user can provide a maf that has already been subset to the region(s) of interest with `mutation_maf`.
#' As a third alternative, the regions can also be specified as a bed file with `bed`.
#' Lastly, this function has a variety of parameters that can be used to further customize the returned plot in many different ways.
#' Refer to the parameter descriptions, examples as well as the vignettes for more demonstrations how this function can be called.
#'
#' @param mutations_maf A data frame containing mutations (MAF format) within a region of interest (i.e. use the get_ssm_by_region).
#' @param metadata should be a data frame with sample_id as a column.
#' @param exclude_classifications Optional argument for excluding specific classifications from a metadeta file.
#' @param drop_unmutated Boolean argument for removing unmutated sample ids in mutated cases.
#' @param classification_column The name of the metadata column to use for ordering and colouring samples.
#' @param bed Optional data frame specifying the regions to annotate (required columns: start, end, name).
#' @param region Genomic region for plotting in bed format.
#' @param custom_colours Provide named vector (or named list of vectors) containing custom annotation colours if you do not want to use standartized pallette.
#' @param hide_ids Boolean argument, if TRUE, ids will be removed.
#'
#' @return ggplot2 object.
#'
#' @import dplyr ggplot2
#' @export
#'
#' @examples
#' #basic usage
#' region = "chr6:90975034-91066134"
#' metadata = get_gambl_metadata()
#' plot = ashm_rainbow_plot(metadata = metadata, region = region)
#'
#' #advanced usages
#' mybed = data.frame(start = c(128806578,
#'                              128805652,
#'                              128748315),
#'                    end = c(128806992,
#'                            128809822,
#'                            128748880),
#'                    name = c("TSS",
#'                             "enhancer",
#'                             "MYC-e1"))
#'
#' ashm_rainbow_plot(mutations_maf = my_mutations,
#'                   metadata = my_metadata,
#'                   bed = mybed)
#'
```

- [ ] My function uses a library that isn't already a dependency of GAMBLR and I made the package aware of this dependency using the function documentation `import` statement. 

Example:
```
#' @return nothing
#' @export
#' @import tidyverse ggrepel
```

## Checklist for changes to existing code

- [ ] I added/removed arguments to a function and updated documentation for all changed/new arguments

- [ ] I tested the new code for compatibility with existing functionality in the Master branch (please provide a reprex of how you tested the original functionality)

### CODE REVIEW

**Comment by mattssca** [comment]
Thanks for working on this @HoumanLM. 

I just casually wanted to point out that this PR adds some real heavyweights in terms of package dependencies (GenomicRanges, IRanges, and Rsamtools). [In the past](https://github.com/morinlab/GAMBLR/pull/115#discussion_r941664185), we wanted to avoid some (or all?) of these. 

It might be the case that we indeed need all of these packages for the function added in this PR to work. But nonetheless, I wanted to bring it up for discussion.

**Comment by mattssca** [comment]
I just had a look at this PR to see why it was failing the build check under Git actions. Not sure if you had a look at this output yet, but this is why the build check failed:

```
Undocumented arguments in documentation object 'findMotif'
  ‘maf’ ‘motif’ ‘projection’ ‘fastaPath’
Documented arguments not in \usage in documentation object 'findMotif':
  ‘maf:’ ‘motif:’ ‘projection:’ ‘fastaPath:’
```

After looking at the documentation for the new function added in this PR, you want to remove the colons after the parameter names, since the string following `@param` needs to be an exact match of the used parameter name inside the function. Does this make sense?

```
#' @param maf: MAF data frame (required columns: Reference_Allele, Chromosome, Start_Position, End_Position)
#' @param motif: The motif sequence (default is WRCY)
#' @param projection: The genome build projection for the variants you are working with (default is grch37)
#' @param fastaPath: Can be a path to a FASTA file
```

Thanks!

**Comment by mattssca** [comment]
Nevermind my last comment, you resolved it in your commit that was published one minute earlier than my review comment. Thanks!

**Comment by HoumanLM** [comment]
Thanks Adam for your explanation.

**Review by Kdreval — APPROVED** [review]

File: R/annotation.R [hunk 1]
```diff
@@ -479,3 +479,200 @@ annotate_sv = function(sv_data,
...
+#' @rawNamespace import(GenomicRanges, except = c("merge", "shift", "union", "intersect", "setdiff", "reduce"))
+#' @import Rsamtools readr dplyr
+#' @export
+#'
+#' @examples
+#' findMotif(maf = maf,
+#'           motif = "WRCY",
+#'           projection = "hg38"
+#'          )
+
```
Comment (rdmorin) [inline R/annotation.R:512]: Name suggestion. Either of these is more descriptive than the current function name

`annotate_mutation_motif_context`

`annotate_ssm_motif_context`

**Review by rdmorin — CHANGES_REQUESTED** [review]
Please rename the new function as per one of my suggestions

File: R/annotation.R (hunk 1 above)
Comment (HoumanLM) [reply R/annotation.R:512]: Sure! I'm going to change it to `annotate_ssm_motif_context`. Thanks!

Comment (HoumanLM) [reply R/annotation.R:512]: Comments were addressed and the function now has the new name.

**Review by rdmorin — APPROVED** [review]
Nice addition to GAMBLR!

**Pull request merged**